### PR TITLE
Add video picker component

### PR DESCRIPTION
**Pull request by kispali**
Add Video Picker component with the contribution of @tkotosz

### CODE REVIEW

File: src/MX/WidgetComponent/Block/Adminhtml/Component/VideoPicker/Uploader.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public function __construct(
+        Context $context,
+        Size $fileSize,
+        Storage $videoStorage,
+        array $data = []
+    ) {
+        parent::__construct($context, $fileSize, $videoStorage, $data);
+
+        $type = $this->_getMediaType();
```
Comment (tkotosz) [inline src/MX/WidgetComponent/Block/Adminhtml/Component/VideoPicker/Uploader.php]: unused variable, can be removed

File: src/MX/WidgetComponent/Form/Component/MediaPicker/Video/Preview.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * @param  string $content
+     * @param  string $videoUrl
+     *
+     * @return string
+     */
+    private function createATag($videoUrl, $content)
+    {
+
```
Comment (tkotosz) [inline src/MX/WidgetComponent/Form/Component/MediaPicker/Video/Preview.php]: empty lines

File: src/MX/WidgetComponent/etc/di.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    <type name="Magento\Cms\Model\Wysiwyg\Images\Storage">
+        <arguments>
+            <argument name="extensions" xsi:type="array">
+                <item name="media_allowed" xsi:type="array">
+                    <item name="mp4" xsi:type="number">1</item>
+                </item>
+            </argument>
+        </arguments>
+    </type>
+</config>
```
Comment (tkotosz) [inline src/MX/WidgetComponent/etc/di.xml]: missing line break

File: src/MX/WidgetComponent/Block/Adminhtml/Component/VideoPicker/Uploader.php (hunk 1 above)
Comment (kispali) [reply src/MX/WidgetComponent/Block/Adminhtml/Component/VideoPicker/Uploader.php]: Sorted

File: src/MX/WidgetComponent/Form/Component/MediaPicker/Video/Preview.php (hunk 2 above)
Comment (kispali) [reply src/MX/WidgetComponent/Form/Component/MediaPicker/Video/Preview.php]: Sorted

File: src/MX/WidgetComponent/etc/di.xml (hunk 3 above)
Comment (kispali) [reply src/MX/WidgetComponent/etc/di.xml]: Sorted

File: src/MX/WidgetComponent/etc/widget.xml [hunk 4]
```diff
@@ -77,6 +77,23 @@
...
+                        <label translate="true">Default Template</label>
+                    </option>
+                </options>
+            </parameter>
+            <parameter name="video_url" xsi:type="block" visible="true" required="true" sort_order="10">
+                <label translate="true">Video</label>
+                <block class="MX\WidgetComponent\Block\Adminhtml\Component\VideoPicker"/>
+            </parameter>
+        </parameters>
+    </widget></widgets>
```
Comment (tkotosz) [inline src/MX/WidgetComponent/etc/widget.xml]: missing line break before </widgets>

Comment (kispali) [reply src/MX/WidgetComponent/etc/widget.xml]: sorted

**Pull request merged**